lib/model.d.ts

```diff
@@ -325,7 +325,7 @@ export interface IncludeOptions {
   /**
    * Load further nested related models
    */
-  include?: Includeable[];
+  include?: Includeable[] | IncludeAll[];
```

Comment:
Oh, I see. Yeah, the typing was defined with `{all: true}`, the `tsc` error apparently changes it to `{all: boolean}`.

Regarding the input: I deliberately used some invalid input to make sure the typing works correctly. As you've mentioned, having `{ all: true, include: ...... }` should never be allowed, but `tsc` accepts it. Here's a more detailed version of what I've tried:

include.ts:
```TypeScript
MyModel.findAll({
  include: [{
    all: true,
    include: [{
      model: AssociatedModel,
      where: { state: Sequelize.col('project.state') },
      limit: 1,
      separate: true,
      order: [['id', 'DESC']]
    }]
  }]
});
``` 
model.d.ts:
```TypeScript
export type Includeable = Model | Association | IncludeOptions | { all: true };

export interface IncludeOptions {
  ...
  include?: Includeable[];
  ...
}
```

The test-case above should fail, but running `npm run build && npm run test` results in no errors whatsoever.